### PR TITLE
👷 Update github workflow configs

### PR DESCRIPTION
**Pull request by yogyagamage**

### CODE REVIEW

**Comment by yogyagamage** [comment]
@algomaster99 can you please review and approve this? 

This PR 
 - Updates the maven version to 3.9.6 in workflow files
 - Excludes automated updates from renovate for maven - this was done before, but was not working
 - Fixes qodana warnings
 - Excludes not-fixable qodana warnings in qodana configs - this was also done before, but was not working
 
 The failing 2 actions cannot be fixed because,
  - check-lockfile action- https://github.com/marketplace/actions/maven-lockfile#:~:text=for%20more%20information.-,%E2%9A%A0%EF%B8%8F,the%20changed%20lockfile%20does%20not%20work%20for%20pull%20requests%20from%20forks.,-See%20https%3A//github
  - build-and-push-action - I do not have permission to push to https://maven.pkg.github.com/chains-project/maven-lockfile (I think that is the reason; the error code is 403-forbidden)

File: maven_plugin/src/main/java/io/github/chains_project/maven_lockfile/data/LockFile.java [hunk 1]
```diff
@@ -32,6 +32,7 @@ public class LockFile {
     private final VersionNumber version;
 
     @SerializedName("lockFileVersion")
+    @SuppressWarnings("FieldMayBeFinal")
     private int lockfileVersion = 1; // TODO: we normally should create an enum with Name -> Numbers
```
Comment (algomaster99) [inline maven_plugin/src/main/java/io/github/chains_project/maven_lockfile/data/LockFile.java:36]:
```suggestion
    private final int lockfileVersion = 1; // TODO: we normally should create an enum with Name -> Numbers
```

**Review by algomaster99 — CHANGES_REQUESTED** [review]
Thanks, @yogyagamage ! Could you try to submit this pull request from the branch in this repository itself? I think that should solve both the problems.

The changes look good to me. Just left a minor change.

File: maven_plugin/src/main/java/io/github/chains_project/maven_lockfile/data/LockFile.java (hunk 1 above)
Comment (yogyagamage) [reply maven_plugin/src/main/java/io/github/chains_project/maven_lockfile/data/LockFile.java:36]: Created a new PR from bug-fix/workflow-fix, and merged it as all the checks passed.

Comment (algomaster99) [reply maven_plugin/src/main/java/io/github/chains_project/maven_lockfile/data/LockFile.java:36]: Sounds good!

**Pull request merged**